lib/logger.dart

```diff
@@ -1,59 +1,96 @@
+import 'package:archive/archive_io.dart';
+import 'package:fimber_io/fimber_io.dart';
+import 'package:flutter_fimber/flutter_fimber.dart';
+import 'package:flutter/foundation.dart';
+import 'package:path_provider/path_provider.dart';
 import 'package:share_plus/share_plus.dart';
 
-final Logger log = Logger('Breez');
+final _log = FimberLog("logger");
 
-Future<File> get _logFile async {
-  var logPath = await ServiceInjector().breezBridge.getLogPath();
-  return File(logPath);
+void shareLog() async {
+  final appDir = await getApplicationDocumentsDirectory();
+  final encoder = ZipFileEncoder();
+  final zipFilePath = "${appDir.path}/breez.logs.zip";
+  encoder.create(zipFilePath);
+  encoder.addDirectory(Directory("${appDir.path}/logs/"));
+  encoder.close();
+  final zipFile = XFile(zipFilePath);
+  Share.shareXFiles([zipFile]);
 }
 
-void shareLog() {
-  _logFile.then((file) {
-    Share.shareFiles(
-      [file.path],
-      text: getSystemAppLocalizations().share_log_text,
-    );
-  });
-}
+class BreezLogger {
+  BreezLogger() {
+    final logLevels = [
+      "V", // log.v Verbose
+      "D", // log.d Debug
+      "I", // log.i Info
+      "W", // log.w Warning
+      "E", // log.e Error
+    ];
+
+    if (kDebugMode) {
+      Fimber.plantTree(FimberTree(
+        logLevels: logLevels,
+      ));
+    }
 
-Future shareFile(String filePath) {
-  return Share.shareFiles(
-    [filePath],
-    text: getSystemAppLocalizations().share_file_title,
-  );
+    getApplicationDocumentsDirectory().then(
+          (appDir) {
+        _pruneLogs(appDir);
+        final tokens = [
+          CustomFormatTree.timeStampToken,
+          CustomFormatTree.levelToken,
+          CustomFormatTree.tagToken,
+          CustomFormatTree.messageToken,
+          CustomFormatTree.fileNameToken,
+          CustomFormatTree.lineNumberToken,
+          CustomFormatTree.filePathToken,
+          CustomFormatTree.exceptionMsgToken,
+          CustomFormatTree.exceptionStackToken,
+        ];
+        Fimber.plantTree(
+          SizeRollingFileTree(
+            DataSize(megabytes: 10),
+            filenamePrefix: "${appDir.path}/logs/breez.",
+            filenamePostfix: ".log",
+            logLevels: logLevels,
+            logFormat: tokens.fold("", (p, e) => p == "" ? e : "$p :: $e"),
+          ),
+        );
+        FlutterError.onError = (FlutterErrorDetails details) async {
+          FlutterError.presentError(details);
+          final name = details.context?.name ?? "FlutterError";
+          final exception = details.exceptionAsString();
+          _log.e("$exception --$name", ex: details, stacktrace: details.stack);
+        };
+      },
+    );
+  }
 }
 
-class BreezLogger {
-  BreezLogger() {
-    BreezBridge breezBridge = ServiceInjector().breezBridge;
-    Logger.root.level = Level.ALL;
-    Logger.root.onRecord.listen((LogRecord rec) {
-      breezBridge.log(rec.message, rec.level.name);
-      print(rec.message);
```

Comment:
Since native loggers are no longer used I think they should be cleaned up.

Reply:
I'm trying to understand which native loggers you are talking about @erdemyerebasmaz 

I'm not changing the java/swift or the golang logs here, only the flutter logs.

Reply:
Okay, I'm still trying to understand how our logger used to work and I may be wrong, so let me try to break things into parts.

First, the cleanup I was talking about:
```
 void log(String msg, String level) {
    _invokeMethodImmediate("log", {"msg": msg, "lvl": level});
  }
```
from BreezBridge is removed and I believe they were handled by:
`public void onMethodCall(MethodCall call, MethodChannel.Result result) {...}` of **Breez.java** 
&
 `func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {...}` of **Breez.swift**.

Since `_breezLib.log()` is removed, their handlers should also be removed. Aren't they?
***
My other question is about logs from native classes such as **ChainSync.java**, **BackgroundTask.swift**.

I believe their logs were added to `{workingDir.path}/logs/bitcoin/mainnet/lnd.log` which was the file we used to share. These are no longer part of our logs, right?
Which I believe the same goes for Go library logs. They can be seen on the LogCat screenshot you've sent but won't be logged through Fimber and the shared log file will be missing native & library logs from how I see it.

Looks like it's been 4 years+ now but maybe @roeierez can clarify things as he was the one implemented logging.

Reply:
The first point you are right, the log method can be safely deleted as we era not using it anymore.

Regarding the second question, the logs are split, there is the `lnd.log` which contains the go and native logs and the `breez.log` which contains the flutter logs, they show up on logcat together but right now they are being saved in different files.

Reply:
Let's log to a single file if we can. If all else fails we can zip and share both files.

Reply:
I can just copy the c-breez strategy and broadcast the logs of the native side to flutter, but I need to change how the library works today.